src/components/Common/UI/ProgressBreadcrumbs/ProgressBreadcrumbsTypes.ts

```diff
@@ -0,0 +1,62 @@
+import type { OnEventType } from '@/components/Base/useBase'
+import type { EventType } from '@/types/Helpers'
+
+export interface Breadcrumb {
+  /**
+   * Unique key for the breadcrumb step
+   */
+  key: string
+  /**
+   * Translation key for the breadcrumb label
+   */
+  label: string
+  /**
+   * Optional translation namespace
+   */
+  namespace?: string
+  /**
+   * Optional variables for the breadcrumb label
+   */
+  variables?: Record<string, unknown>
+  /**
+   * Event handler for breadcrumb navigation
+   */
+  onNavigate?: (context: unknown) => unknown
```

Comment:
Ok, so with adapters we'll have to think about how the partner would wire this up 🤔 I'm thinking maybe we could pass the key/id here? That is already configuring a unique identifier similar to what we do with tabs. The partner would know they just need to provide that key/id as an argument for whatever click handling they configure